src/compiler/program.ts

```diff
@@ -2413,7 +2413,8 @@ export function createProgram(rootNamesOrOptions: readonly string[] | CreateProg
     }
 
     function getMergedBindAndCheckDiagnostics(sourceFile: SourceFile, includeBindAndCheckDiagnostics: boolean, ...allDiagnostics: (readonly Diagnostic[] | undefined)[]) {
-        const flatDiagnostics = flatten(allDiagnostics);
+        const flatDiagnostics = Array.from(sortAndDeduplicateDiagnostics(flatten(allDiagnostics)));

```

Comment:
Instead this needs to be in getDiagnosticsHelper where we return diagnostics for just one file.
https://github.com/microsoft/TypeScript/blob/main/src/compiler/program.ts#L2641

We want to avoid having to sort through diagnostics for each file if they will be combined and then sorted.

Reply:
Thanks @sheetalkamat for the review. I've moved this to getDiagnosticsHelper, synced up with main and accept a new baseline for a test case where the diagnostics are now ordered.